code/modules/urist/modules/GOAI/_datastructures/faction_data.dm

```diff
@@ -0,0 +1,69 @@
+/*
+// This is the actual data for a faction (a container for
+// stuff like relationships, members and whatever else).
+//
+// The faction AI should NOT have stuff like this directly
+// hardcoded anywhere - if it does, we cannot have factions
+// that are NOT driven by AI (and we might need it for stuff
+// like PC factions, where we want third-party AIs to access
+// this data without having an AI loop for the PC faction itself.
+*/
+
+/datum/faction_data
```

Comment:
i haven't had a chance to fully look through everything, but this is something where you could hook into an existing framework I made (the faction subsystem), and expand on it to fit this quite easily. See https://github.com/UristMcStation/UristMcStation/blob/master/code/controllers/subsystems/factions.dm for what i mean

Reply:
It's already low-key integrated: https://github.com/UristMcStation/UristMcStation/pull/1090/files#diff-60c5f1f08ce58a907d274c4d3df54bae697ab4f96fdd097a35389232dfb8436dR99-R101

Beyond that, I'm trying to keep modules decoupled - the design here is that a basic framework should be generic and have 'drivers' into SS13 code as an optional layer on top. 

So, a proper Faction AI can hook into that system, while generic AI might not; an AI might be tied to any DM object, or be completely abstract, and so on.

This is quite nice because it makes the GOAI code portable between this PR and the GOAI repo with just a couple of ifdefs switching between SS13 integrations and dev stubs.